### PR TITLE
fix(SFINT-2500): Add ts-es5-istanbul-coverage loader

### PR DESCRIPTION
**Pull request by louis-bompart**
This package will add the istanbul ignore statement where they need to be to circumvent the issue with TypeScript transpilation to ES5.
This will only apply when the code is transpiled for unit tests, not for production.
Also remove all the istanbul ignore comments now that they're not needed anymore.

### CODE REVIEW

File: config/webpack.config.karma.js [hunk 1]
```diff
@@ -3,21 +3,42 @@ const path = require('path');
...
-    outDir: undefined
-};
-webpackConfig.module.rules.push({
-    enforce: 'post',
-    test: /\.ts$/,
-    loader: 'istanbul-instrumenter-loader',
-    exclude: path.resolve('tests/'),
-    query: {
-        esModules: true
+webpackConfig.module.rules = [
```
Comment (jeremierobert-coveo) [inline config/webpack.config.karma.js]: Pourquoi tu push plus la config ?

File: config/webpack.config.karma.js [hunk 2]
```diff
@@ -3,21 +3,42 @@ const path = require('path');
...
+    {
+        test: /\.ts$/,
+        use: [
+            {
+                loader: 'ts-es5-istanbul-coverage'
```
Comment (jeremierobert-coveo) [inline config/webpack.config.karma.js]: Le premier loader deviens l'instrumenteur ?

File: config/webpack.config.karma.js (hunk 1 above)
Comment (louis-bompart) [reply config/webpack.config.karma.js]: Car ca devenait complexe pour pas vraiment grand chose imo

File: config/webpack.config.karma.js (hunk 2 above)
Comment (louis-bompart) [reply config/webpack.config.karma.js]: Check: https://www.npmjs.com/package/ts-es5-istanbul-coverage
tl;dr: it runs second. so it instrument the instrumented
